w2_d2_1_underscore_templating/README.md

```diff
@@ -0,0 +1,174 @@
+
+  ```html
+  <body>
+    <div class="container">
+      <div class="row">
+        <div class="col-md-6">
+          <div id="pets-list">
+        </div>
+      </div>
+    </div>
+  </body>
+  ```
+
+4. Compile your template in your `script.js` file. This step turns `petsTemplate` into a function, where we will later pass in the data we want to render in the template.
+
+  ```js
+  var petsTemplate = _.template($('#pets-template').html());
+  ```
+
+5. Set up some test data in your `script.js` file. **Your object keys must match the parameters you set up in your template!** (e.g. `<%= name %>` and `<%= species %>`)
+
+  ```js
+  var pets = [
+    {name: "Sprinkles", species: "cat"},
+    {name: "Bagel", species: "dog"},
+    {name: "Fluffy", species: "dinosaur"}
+  ];
+  ```
+
+6. Iterate through your test data, using your template to create a new HTML element for each data object. Once the element is created, append it to the list element you set up in step 3 (e.g. `pets-list`). **Note:** We're also saving the object's index in the array to an HTML data attribute (this will come in handy when we want to delete elements from the DOM).
+
+  ```js
+  _.each(pets, function (pet, index) {
+    var $pet = $(petsTemplate(pet));
+    $pet.attr('data-index', index);
+    $petsList.append($pet);
+  });
+  ```
+
+## Challenges, Part 1
+
+1. Add the Underscore CDN to your To Do app. Make sure to require it before your custom script.
+
+2. Create an underscore template below your custom script. Remember to give it an id. Think about the data you will be displaying related to each to do item. That data will determine your template's parameters (inside the `<%= %>` tags).
+
+3. Set up a "list" element in your HTML where you will eventually append the data from your template. A `<div>` or a `<ul>` is a good choice. Remember to give it an id.
+
+## Challenges, Part 2
+
+1. Compile your template, using jQuery to select it by its id.
+
+2. Set up test data in your custom script file. This should be an array of objects, where the keys match your template's paraeters.
+
+## Challenges, Part 3
+
+1. Iterate through your test data, creating a new HTML element for each object in the array and appending it to the "list" element you set up in part 1 of the challenges.
+
+2. Open your `index.html` in the browser to see if your test data displays on the page. If not, check the JS console to see what errors you're getting.
+
+3. Once you've successfully displayed your test data on the page, refactor your new To Do form to use the template when adding new tasks.
+
+## Stretch Challenges
+
+1. When a new task is created, make sure you are pushing it into the array of "test" data you set up in part 2 of the challenges to keep your model updated with your view.
+
+2. Implement a delete functionality to remove tasks from your To Do list. Think about the places you'll need to make updates (**Hint:** Model and View).
```

Comment:
We'll be touching on it this afternoon, but I could definitely add in more info about that to make it more clear!